benchmark/src/main/java/com/google/instrumentation/trace/HttpPropagationUtilBenchmark.java

```diff
@@ -0,0 +1,109 @@
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+/** Benchmarks for {@link HttpPropagationUtil}. */
+@State(Scope.Benchmark)
+public class HttpPropagationUtilBenchmark {
+  private static final byte[] traceIdBytes =
+      new byte[] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'a'};
+  private byte[] spanContextBytes;
+  private String spanContextString;
+  private static final TraceId traceId = TraceId.fromBytes(traceIdBytes);
+  private static final byte[] spanIdBytes = new byte[] {(byte) 0xFF, 0, 0, 0, 0, 0, 0, 0};
+  private static final SpanId spanId = SpanId.fromBytes(spanIdBytes);
+  private static final byte[] traceOptionsBytes = new byte[] {0, 0, 0, 1};
+  private static final TraceOptions traceOptions = TraceOptions.fromBytes(traceOptionsBytes);
+  private SpanContext spanContext;
+  private String spanContextStringHttp;
+
+  @Setup
+  public void setUp() throws Exception {
+    spanContext = new SpanContext(traceId, spanId, traceOptions);
+    spanContextStringHttp = HttpPropagationUtil.toHttpHeaderValue(spanContext);
+    spanContextBytes = new byte[16 + 8 + 4];
+    traceId.copyBytesTo(spanContextBytes, 0);
+    spanId.copyBytesTo(spanContextBytes, 16);
+    traceOptions.copyBytesTo(spanContextBytes, 24);
+    spanContextString = BaseEncoding.base16().lowerCase().encode(spanContextBytes);
+  }
+
+  /**
+   * This benchmark attempts to measure performance of {@link
+   * HttpPropagationUtil#toHttpHeaderValue(SpanContext)}.
+   */
+  @Benchmark
+  @BenchmarkMode(Mode.SampleTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public String toHttpHeaderValueSpanContext() {
+    return HttpPropagationUtil.toHttpHeaderValue(spanContext);
+  }
+
+  /**
+   * This benchmark attempts to measure performance of {@link
+   * HttpPropagationUtil#fromHttpHeaderValue(CharSequence)}.
+   */
+  @Benchmark
+  @BenchmarkMode(Mode.SampleTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public SpanContext fromHttpHeaderValueSpanContext() {
+    return HttpPropagationUtil.fromHttpHeaderValue(spanContextStringHttp);
+  }
+
+  /**
+   * This benchmark attempts to measure performance of {@link
+   * HttpPropagationUtil#toHttpHeaderValue(SpanContext)} then {@link
+   * HttpPropagationUtil#fromHttpHeaderValue(CharSequence)}.
+   */
+  @Benchmark
+  @BenchmarkMode(Mode.SampleTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public SpanContext toFromHttpFormatSpanContext() {
+    return HttpPropagationUtil.fromHttpHeaderValue(
+        HttpPropagationUtil.toHttpHeaderValue(spanContext));
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.SampleTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public String base16EncodeSpanContext() {
+    return BaseEncoding.base16().lowerCase().encode(spanContextBytes);
```

Comment:
Why do the last three benchmarks test base-16 encoding and decoding?  Is that because it is part of the implementation of one of the other methods that operates on `SpanContext`?  I think it's worth a comment.